packages/theme-check-common/src/checks/valid-render-snippet-params/index.ts

```diff
@@ -0,0 +1,135 @@
+      recommended: true,
+      url: 'https://shopify.dev/docs/storefronts/themes/tools/theme-check/checks/valid-render-snippet-params',
+    },
+    type: SourceCodeType.LiquidHtml,
+    severity: Severity.WARNING,
+    schema: {},
+    targets: [],
+  },
+
+  create(context) {
+    function partitionParameters(
+      liquidDocParameters: Map<string, LiquidDocParameter>,
+      providedParams: LiquidNamedArgument[],
+    ) {
+      const providedParameters = new Map(providedParams.map((arg) => [arg.name, arg]));
+
+      const missingRequiredParams: LiquidDocParameter[] = [];
+      const unknownProvidedParams: LiquidNamedArgument[] = [];
+
+      // Check required params
+      for (const param of liquidDocParameters.values()) {
+        if (param.required && !providedParameters.has(param.name)) {
+          missingRequiredParams.push(param);
+        }
+      }
+
+      // Check provided params
+      for (const arg of providedParameters.values()) {
+        const liquidDocParamDef = liquidDocParameters.has(arg.name);
+        if (!liquidDocParamDef) {
+          unknownProvidedParams.push(arg);
+        }
+      }
+
+      return {
+        missingRequiredParams,
+        unknownProvidedParams,
+      };
+    }
+
+    function reportMissingParams(
+      missingRequiredParams: LiquidDocParameter[],
+      node: RenderMarkup,
+      snippetName: string,
+    ) {
+      for (const param of missingRequiredParams) {
+        context.report({
+          message: `Missing required parameter '${param.name}' in render tag for snippet '${snippetName}'`,
+          startIndex: node.position.start,
+          endIndex: node.position.end,
+          suggest: [
+            {
+              message: `Add required parameter '${param.name}'`,
+              fix: (fixer) => {
+                // This will insert the argument at the end position
+                return fixer.insert(
+                  node.position.end - 1,
+                  `, ${param.name}: ${getDefaultValueForType(param.type)}`,
+                );
+              },
+            },
+          ],
+        });
+      }
+    }
+
+    function reportUnknownParams(
+      unknownProvidedParams: LiquidNamedArgument[],
+      snippetName: string,
+    ) {
+      for (const arg of unknownProvidedParams) {
+        context.report({
+          message: `Unknown parameter '${arg.name}' in render tag for snippet '${snippetName}'`,
+          startIndex: arg.position.start,
+          endIndex: arg.position.end,
+          suggest: [
+            {
+              message: `Remove '${arg.name}'`,
+              fix: (fixer) => {
+                return fixer.remove(arg.position.start, arg.position.end);
+              },
+            },
+          ],
+        });
+      }
+    }
+
+    return {
+      async RenderMarkup(node: RenderMarkup) {
+        if (!isLiquidString(node.snippet) || node.variable) {
```

Comment:
To avoid breaking the check for existing themes, we are
1) Only reporting if a `liquidDoc` definition exists **with @params**
2) Limiting reporting variables passed down - [ref](https://shopify.dev/docs/api/liquid/tags/render#render-passing-variables-to-a-snippet), so excluding `with` and `for`, at least for now. 

_*2: AFAIK, You can only pass a single value using this syntax, so there's little value in adding LiquidDoc imo. Low priority but still nice to have_